docs/rfcs/XeGPU.md

```diff
@@ -171,122 +142,341 @@ When the input matrix is a lower-precision data type (lower than 32bit), the inp
+
+## XeGPU Attributes to support Work Item Level semantics  
+
+Attribute `xegpu.sg_map` describes the mapping between work item (WI) and the 2D tensor specified by the tensor descriptor. To distribute the XeGPU operation to work items, the tensor_desc must be specified with the `sg_map` attribute at the tensor description creation time.
+
+Within the sg_map, `wi_layout` specifies the layout of work items, describing the mapping of work items to the tensor. wi_layout[0] x wi_layout[1] must be equal to the total number of work items within a subgroup. `wi_data` specifies the data elements assigned to each work item for a single distribution. In the example below, the subgroup has 16 work items in wi_layout=[1, 16], each accessing 1 element as specified by wi_data=[1,1]. 
+
+```mlir
+  #sg_map_a = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  %tdesc_a = xegpu.create_nd_tdesc %mem_addr, %offsets:2, %base_shape:2,%base_stride:2
+		: uint64, index, index, index, index, index, index
+     	into tensor_desc<8x16xbf16, #sg_map_a>
+```
+
+wi_data_size refers to the data size mapped to indiviudal work item, and sg_map_size to the collective size by all the work items as specified by sg_map. sg_map_size represents the minimun size of 2D tensor to be distributed to work items in a subgroup. tensor_size refers to the size of the tensor sepcified by tensor_desc. 
+In the example above, wi_data_size is 1, sg_map_size is 16, tensor_size is 128. 
+```mlir
+        wi_data_size = wi_data[0] x wi_data[1]
+	subgroup_size == wi_layout[0] x wi_layout[1] 
+ 	sg_map_size = subgroup_size * wi_data_size 
+  	tensor_size = tensor_desc[0] x tensor_desc[1]
+```
+
+Size distribution rule can be represented as following: tensor_desc[0] must be divisible by wi_layout[0] x wi_data[0], tensor_desc[1] by wi_layout[1] x wi_data[1], and tensor_size by sg_map_size. The 2D subtensor is evenly distributed to work items, so each work item gets a 2D data fragments. 
+
+The size of the result data fragement per work item can be computed by the following: 
+```mlir
+	WI_data_frag[0] = tensor_desc[0]/wi_layout[0]
+	WI_data_frag[1] = tensor_desc[1]/wi_layout[1]
+```
+
+The WI dsitribution is represented by the shape of the result vector being loaded, which is being reduced from [tensor_decs[0], tensor_desc[1]] to [WI_data_frag[0], WI_data_frag[1]].
+
+With `sg_map` attribute attached to tensor_desc, xegpu.load_nd operates in SIMT flavor and returns back a fragement associated with individual work item. The tensor_desc in the example below specifies a tensor of 8x16 elements, which is decomposed to 8x1 subtensors, each with sg_map_size 1x16. The result vector <8x16xbf16> is loaded and distributed to each WI as <8x1xbf16>.  
+```mlir
+  #sg_map_a = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  %vector_a = xegpu.load_nd %tdesc_a:
+     tensor_desc<8x16xbf16, #sg_map_a> into vector<8x1xbf16>
```

Comment:
btw do we need to attach sg_map_a here? it is already attached to tdesc_a bit earlier

Reply:
Yes, it wouldn't be the same type without the attribute.